.pre-commit-config.yaml

```diff
@@ -11,20 +12,22 @@ repos:
       - id: end-of-file-fixer
       - id: mixed-line-ending
       - id: trailing-whitespace
-        exclude: \.md$
+        exclude: \.md|.svg|.ipynb$
       - id: no-commit-to-branch
   - repo: https://github.com/astral-sh/ruff-pre-commit
     rev: v0.11.12
     hooks:
       - id: ruff-check
+        types_or: [ python, pyi ]
         args: [--fix]
       - id: ruff-format
+        types_or: [ python, pyi ]
 
   - repo: local
     hooks:
       - id: ty
         name: ty check
-        entry: uvx ty check .
+        entry: uvx ty check . --exclude docs/tutorials/*.ipynb
```

Comment:
> [!TIP]
> Here, since we used the `pass_filenames` it means it runs on every file that we have so later we can't use the `exclude` argument in the pre-commit configuration; therefore I need to exclude the `docs/` directory directly.